src/components/Toolbar/Index.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+    StyleSheet,
+    ToolbarAndroid,
+    ToolbarAndroidAction
+} from 'react-native';
+import * as React from 'react';
+import icons, {
+    images
+} from '../../assets/index';
+
+/*Only one button is permitted on the left
+it must either be a menu button which has a 'burger' icon
+or a back button which has an arrow icon*/
+export enum LeftButtonTypes {
+    menu = "menu",
+    back = "back"
+}
+/*Multiple buttons can be added to the right side of the toolbar
+if an icon isn't specified the title will be displayed in its place*/
+interface RightButtonType {
+    title: string;
+    icon ? : any;
+    action: () => void;
+}
+/*This specifies the toolbar properties we intend to allow the pages to control
+currently the title, left button type (menu or back) and right buttons can be specified*/
+interface ToolbarProps {
+    title: string;
+    leftButtonType: LeftButtonTypes;
+    rightButtons: Array < RightButtonType > ;
+}
+
+export default class Toolbar extends React.Component < ToolbarProps, {} > {
+    constructor(props: ToolbarProps) {
+    super(props);
+    }
+    /*When a button is selected by a user, find its position 
+    in the actions array and call specified function*/
+    onActionSelected = (position: number) => {
+        this.props.rightButtons[position].action();
+    }
+    render() {
+        //array to hold all action (right buttons) for the page
+        let actionArray = new Array < ToolbarAndroidAction > ();
+        /*Iterate through right buttons added to the page
```

Comment:
I think we don't need these comments explaining what is happening

Reply:
I think comments explaining how functions work are helpful when you come back to code later on.

Reply:
Sometimes yeah, but I would agree with @beaesguerra that sometimes focusing more on making sure your code is self explanatory is better than over-commenting